pkg/image/vulnerability_scan.go

```diff
@@ -0,0 +1,142 @@
+
+	trivyArgs := []string{"image", "--quiet", "--format", "json"}
+	if imageinDir {
+		trivyArgs = append(trivyArgs, "--input", imagePath)
+	} else {
+		trivyArgs = append(trivyArgs, imagePath)
+		authParams := getAuthStringForTrivyScan(auth, insecure)
+		trivyArgs = append(trivyArgs, authParams...)
+	}
+	if settings.Ignore != nil {
+		if settings.Ignore.Severity != nil {
+			severity := getSeverityStringForTrivyScan(*settings.Ignore.Severity)
+			trivyArgs = append(trivyArgs, "--severity", severity)
+		}
+		if settings.Ignore.Unfixed != nil {
+			trivyArgs = append(trivyArgs, "--ignore-unfixed", strconv.FormatBool(*settings.Ignore.Unfixed))
+		}
+	}
+
+	cmd := exec.CommandContext(ctx, "trivy", trivyArgs...)
+
+	// Print the command to be executed
+	log.Println(cmd.String())
+
+	cmd.Stdin = nil
+
+	result, err := cmd.CombinedOutput()
+	if err != nil {
+		log.Printf("failed to run trivy:\n%s", string(result))
+		return nil, fmt.Errorf("failed to run trivy: %w", err)
+	}
+
+	var trivyResult TrivyResult
+	if err := json.Unmarshal(result, &trivyResult); err != nil {
+		return nil, err
+	}
+
+	// Get the vulnerabilities from the trivy scan
+	vulnerabilities := parseTrivyResult(trivyResult, settings.Ignore)
+
+	// Sort the vulnerabilities by severity
+	severityOrder := map[string]int{"CRITICAL": 0, "HIGH": 1, "MEDIUM": 2, "LOW": 3, "UNKNOWN": 4}
+	sort.Slice(vulnerabilities, func(i, j int) bool {
+		return severityOrder[vulnerabilities[i].Severity] < severityOrder[vulnerabilities[j].Severity]
+	})
+
+	if len(vulnerabilities) > vulnCountLimit {
+		vulnerabilities = vulnerabilities[:vulnCountLimit]
+	}
+
+	return vulnerabilities, nil
+}
+
+func getSeverityStringForTrivyScan(ignoreSeverities string) string {
+	switch strings.ToLower(ignoreSeverities) {
+	case "low":
+		return "MEDIUM,HIGH,CRITICAL"
+	case "medium":
+		return "HIGH,CRITICAL"
+	case "high":
+		return "CRITICAL"
+	default:
+		return "LOW,MEDIUM,HIGH,CRITICAL"
+	}
+}
+
+func parseTrivyResult(trivyResult TrivyResult, ignoreOptions *buildapi.VulnerabilityIgnoreOptions) []buildapi.Vulnerability {
+	var totalVulnerabilities []TrivyVulnerability
+	for _, result := range trivyResult.Results {
+		totalVulnerabilities = append(totalVulnerabilities, result.Vulnerabilities...)
+	}
+
+	var vulnerabilities []buildapi.Vulnerability
+
+	//create a map for ignored vulnerabilities
+	ignoreMap := make(map[string]bool)
+	if ignoreOptions != nil {
+		for _, vuln := range ignoreOptions.ID {
+			ignoreMap[vuln] = true
+		}
+	}
+	for _, vuln := range totalVulnerabilities {
+		if !ignoreMap[vuln.VulnerabilityID] {
+			vulnerability := buildapi.Vulnerability{
+				ID:       vuln.VulnerabilityID,
+				Severity: vuln.Severity,
+			}
+			vulnerabilities = append(vulnerabilities, vulnerability)
+		}
+	}
+	return vulnerabilities
+}
+
+func getAuthStringForTrivyScan(auth *authn.AuthConfig, insecure bool) []string {
+	authParams := []string{}
```

Comment:
Nit: No need to initialize a slice if there is an `append` coming anyway.
```suggestion
	var authParams []string
```